PrinceOfVersionsMacSample/PrinceOfVersionsMacSample/Swift Implementation/AutomaticCheckController.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  AutomaticCheckController.swift
+//  PrinceOfVersionsMacSample
+//
+//  Created by Jasmin Abou Aldan on 20/09/2019.
+//  Copyright © 2019 infinum. All rights reserved.
+//
+
+import Cocoa
+import PrinceOfVersions
+
+class AutomaticCheckController: NSViewController {
+
+    // MARK: - Private properties
+    // MARK: IBOutlets
+
+     @IBOutlet weak var appStateTextField: NSTextField!
+     @IBOutlet weak var metaTextField: NSTextField!
+
+    // MARK: - View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do view setup here.
+        checkAppVersion()
+    }
+}
+
+// MARK: - Private methods -
+
+private extension AutomaticCheckController {
+
+    func checkAppVersion() {
+
+        let princeOfVersionsURL = URL(string: Constants.princeOfVersionsURL)!
+        PrinceOfVersions().checkForUpdates(
+            from: princeOfVersionsURL,
+            newVersion: { [unowned self] (versionData, isMinimumVersionSatisfied, meta) in
```

Comment:
```suggestion
            newVersion: { [weak self] (versionData, isMinimumVersionSatisfied, meta) in
```
Never use unowned on API calls which you cannot cancel or control